nncf/quantization_metric.py

```diff
@@ -0,0 +1,211 @@
+             self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS]
+
+        self.stat[self.MAX_TENSOR_ACTIVATIONS_SHAPE] = max(self._compressed_model.input_infos,\
+             key=lambda k: np.prod(k.shape)).shape
+
+    def get_metric_table(self):
+        table = Texttable()
+        data = [self.header]
+        drow = {h: 0 for h in self.header}
+        for column_name in self.header:
+            drow[column_name] = self.stat[column_name]
+        row = [drow[h] for h in self.header]
+        data.append(row)
+        table.add_rows(data)
+        return table
+
+
+class ShareBlueEdgesCompressedGraph(BaseMetric):
+    COUNT_BLUE_EDGES = 'count blue edges'
+
+    def __init__(self, compressed_model: NNCFNetwork):
+        super().__init__()
+        self._compressed_model = compressed_model
+        self._compressed_graph = self._compressed_model.get_graph()
+        nx.set_edge_attributes(self._compressed_graph._nx_graph, 'black', 'color')
+        self.stat = {}
+
+    def collect(self):
+        self.stat[self.COUNT_BLUE_EDGES] = 0
+        self.header = [self.COUNT_BLUE_EDGES]
+        input_nodes = self._compressed_graph.get_input_nodes()
+        queue = deque()
+        for input_node in input_nodes:
+            next_nodes = self._compressed_graph.get_next_nodes(input_node)
+            for next_node in next_nodes:
+                queue.appendleft(next_node)
+
+        while len(queue) != 0:
+            node = queue.pop()
+            if node.op_exec_context.operator_name == 'symmetric_quantize' \
+                 or node.op_exec_context.operator_name == 'asymmetric_quantize':
+                next_nodes = self._compressed_graph.get_next_nodes(node)
+                for next_node in next_nodes:
+                    edge = self._compressed_graph.get_nx_edge(node, next_node)
+                    edge['color'] = 'blue'
+                    queue.appendleft(next_node)
+                    self.stat[self.COUNT_BLUE_EDGES] += 1
+            elif is_nncf_module(self._compressed_model.get_module_by_scope(node.op_exec_context.scope_in_model)):
+                if node.op_exec_context.scope_in_model in self._compressed_model.get_nncf_module_scopes():
+                    self._coloring_edges(node, queue)
+            else:
+                self._coloring_edges(node, queue)
+
+        self.stat[self.COUNT_BLUE_EDGES] /= len(self._compressed_model.get_graph().get_nx_graph_copy().edges)
+        self.stat[self.COUNT_BLUE_EDGES] *= 100
+
+    def _coloring_edges(self, node, queue):
+        prev_nodes = self._compressed_graph.get_previous_nodes(node)
+        status = True
+        for prev_node in prev_nodes:
+            prev_node_op_name = prev_node.op_exec_context.operator_name
+            if prev_node_op_name not in ('symmetric_quantize', 'asymmetric_quantize'):
+                edge = self._compressed_graph.get_nx_edge(prev_node, node)
+                if edge['color'] != 'blue':
```

Comment:
I see you really count the blue edges here. This is wrong and should be changed. You are supposed to count the CFG edges that correspond to quantized tensors, not edges with a color blue. The blue color is irrelevant everywhere - it only pertains to the visualization of the graph. In no way is the color blue used meaningfully in the actual quantization algorithms.

Reply:
@vshampor  Here I consider that if a quantized tensor passes through an edge, then this is a blue edge. Otherwise, we have no differences between this metric and `NetworkQuantizationShareMetric`.